src/H5FDsubfiling/H5subfiling_common.c

```diff
@@ -940,46 +856,140 @@ init_subfiling(const char *base_filename, uint64_t file_id, H5FD_subfiling_param
      * Setup the application topology information, including the computed
      * number and distribution map of the set of I/O concentrators
      */
-    if (init_app_topology(context_id, subfiling_config, comm, node_comm, &app_topology) < 0)
-        H5_SUBFILING_GOTO_ERROR(H5E_VFL, H5E_CANTINIT, FAIL, "couldn't initialize application topology");
+    if (H5FD__subfiling_init_app_topology(context_id, subfiling_config, comm, node_comm, &app_topology) < 0)
+        HGOTO_ERROR(H5E_VFL, H5E_CANTINIT, FAIL, "couldn't initialize application topology");
+
+    /* Set non-zero fields */
```

Comment:
This function was called `init_subfiling` because it initializes both the context and topology objects. Moving the code out of the `init_subfiling_context` to here just makes this harder to reason about IMO.

Reply:
OK, the comment in the header seemed to imply that init_subfiling was mainly about initializing the context, so I brought up the context initialization into it.   I'll rename this routine to something more general (H5FD__subfiling_setup, possibly) and move all the context initialization back into one routine.